src/test/java/com/solace/connector/kafka/connect/source/SolMessageProcessorIFTest.java

```diff
@@ -0,0 +1,142 @@
+package com.solace.connector.kafka.connect.source;
+
+import static com.solace.connector.kafka.connect.source.SolaceSourceConstants.SOL_SH_APPLICATION_MESSAGE_ID;
+import static com.solace.connector.kafka.connect.source.SolaceSourceConstants.SOL_SH_APPLICATION_MESSAGE_TYPE;
+import static com.solace.connector.kafka.connect.source.SolaceSourceConstants.SOL_SH_CORRELATION_ID;
+import static com.solace.connector.kafka.connect.source.SolaceSourceConstants.SOL_SH_COS;
+import static com.solace.connector.kafka.connect.source.SolaceSourceConstants.SOL_SH_DELIVERY_MODE;
+import static com.solace.connector.kafka.connect.source.SolaceSourceConstants.SOL_SH_DESTINATION;
+import static com.solace.connector.kafka.connect.source.SolaceSourceConstants.SOL_SH_REPLY_TO_DESTINATION;
+import static com.solace.connector.kafka.connect.source.SolaceSourceConstants.SOL_SH_REPLY_TO_DESTINATION_TYPE;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import com.solace.connector.kafka.connect.source.msgprocessors.SolSampleSimpleMessageProcessor;
+import com.solacesystems.common.util.ByteArray;
+import com.solacesystems.jcsmp.BytesXMLMessage;
+import com.solacesystems.jcsmp.DeliveryMode;
+import com.solacesystems.jcsmp.SDTException;
+import com.solacesystems.jcsmp.SDTMap;
+import com.solacesystems.jcsmp.TextMessage;
+import com.solacesystems.jcsmp.User_Cos;
+import com.solacesystems.jcsmp.impl.QueueImpl;
+import com.solacesystems.jcsmp.impl.RawSMFMessageImpl;
+import com.solacesystems.jcsmp.impl.TopicImpl;
+import com.solacesystems.jcsmp.impl.sdt.MapImpl;
+import com.solacesystems.jcsmp.impl.sdt.StreamImpl;
+import java.math.BigInteger;
+import java.util.UUID;
+import org.apache.kafka.connect.header.ConnectHeaders;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class SolMessageProcessorIFTest {
+
+  private SolMessageProcessorIF messageProcessor;
+
+  @BeforeEach
+  void setUp() {
+    messageProcessor = new SolSampleSimpleMessageProcessor();
+  }
+
+  @Test
+  void testUserPropertiesMappingGivenNullUserPropertyMap() {
+    final BytesXMLMessage message = mock(TextMessage.class);
+    when(message.getProperties()).thenReturn(null);
+
+    ConnectHeaders kafkaHeaders = messageProcessor.userPropertiesToKafkaHeaders(message);
+    assertThat("getProperties() is null", kafkaHeaders.isEmpty());
+  }
+
+  @Test
+  void testUserPropertiesMappingGiveEmptyUserPropertyMap() {
+    final SDTMap solMsgUserProperties = new MapImpl();
```

Comment:
Instead of using JCSMP's internal `*Impl` classes, we should try to use the public APIs whenever possible.

Reply:
this goes for all the Impl classes. Not just `Map`. e.g. `QueueImpl`, `TopicImpl`, etc are still there.